src/core/pipe/pipe.build.js

```diff
@@ -0,0 +1,20 @@
+import Middleware from 'core/services/middleware';
+
+export default function build(model, valueFactory) {
+	const pipes = model.data().pipe;
+	const middleware = new Middleware(pipes);
+
+	return function run(source, changes) {
+		const context = {
+			model: model,
+			source: source,
+			valueFactory: valueFactory,
+			changes: changes
+		};
+
+		// TODO: add progress
```

Comment:
Add progress. Maybe provide events `invalidateStart`, `invalidateEnd` or like that?